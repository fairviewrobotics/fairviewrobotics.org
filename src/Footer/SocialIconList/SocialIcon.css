.social-icon {
  height: 1.5em;
  width: 3em;
  cursor: pointer;
  background-size: auto 100%;
  background: no-repeat center center;
  opacity: 1;
  transition: color 0.25s ease;
  color: var(--main-color);
  font-size: 1.65em;
  text-align: center;
}

.social-icon:hover {
  color: var(--second-color);
}