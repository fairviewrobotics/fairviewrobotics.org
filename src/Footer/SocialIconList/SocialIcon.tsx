import React, { PureComponent } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import styles from './SocialIcon.module.css';

export type SocialIconProps = {
  icon: any;
  url: string;
}

export default class SocialIcon extends PureComponent<SocialIconProps> {
  render() {
    return (
      <a 
        className={styles.socialIcon}
        target="_blank"
        rel="noopener noreferrer"
        href={this.props.url}>
        <FontAwesomeIcon icon={this.props.icon} />
      </a>
    );
  }
}
