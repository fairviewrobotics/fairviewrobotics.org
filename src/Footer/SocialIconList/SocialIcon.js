import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';

import './SocialIcon.css';

export default class SocialIcon extends PureComponent {

  static propTypes = {
    icon: PropTypes.any.isRequired,
    url: PropTypes.string.isRequired
  };

  render() {
    return (
      <a className="social-icon"
         target="_blank"
         href={this.props.url}>
        <FontAwesomeIcon icon={this.props.icon} />
      </a>
    );
  }
}
