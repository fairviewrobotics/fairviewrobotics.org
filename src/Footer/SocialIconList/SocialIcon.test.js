import React from "react";
import { shallow } from "enzyme";
import SocialIcon from "./SocialIcon";

describe('SocialIcon', () => {
  let wrapper;

  it('renders without crashing', () => {
    shallow(<SocialIcon icon={{ name: 'icon' }} url="" />)
  });

  describe('given an `icon` and `url`', () => {
    beforeEach(() => {
      wrapper = shallow(
        <SocialIcon icon={{ name: 'I Am A Font Awesome Icon' }} url="https://socialmediasite.com/" />
      );
    });

    it('should render an linked icon', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});