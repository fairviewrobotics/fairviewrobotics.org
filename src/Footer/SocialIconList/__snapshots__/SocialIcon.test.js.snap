// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcon given an \`icon\` and \`url\` should render an linked icon 1`] = `
<a
  className="socialIcon"
  href="https://socialmediasite.com/"
  target="_blank"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "name": "I Am A Font Awesome Icon",
      }
    }
    listItem={false}
    mask={null}
    name=""
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    transform={null}
  />
</a>
`;
